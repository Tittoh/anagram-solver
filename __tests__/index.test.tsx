import { render, screen } from '@testing-library/react';

import Home from '../src/pages/index';
import '@testing-library/jest-dom';

describe('Home', () => {
  it('renders a next.js logo', () => {
    render(<Home />);

    const heading = screen.getByRole('heading', {
      name: /Anagram Solver/i,
    });

    expect(heading).toBeInTheDocument();
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });
});
