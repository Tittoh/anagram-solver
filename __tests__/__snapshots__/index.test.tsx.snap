// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches the snapshot 1`] = `
<DocumentFragment>
  <main
    class="min-h-screen relative pb-20 bg-slate-100 flex flex-col justify-center items-center"
  >
    <section
      class="mx-auto max-w-3xl p-4 lg:px-8 "
    >
      <div
        class="-mt-16"
      >
        <div>
          <h1
            class="text-4xl py-1 font-bold tracking-tight text-center sm:text-6xl bg-gradient-to-r from-indigo-500 via-blue-500 to-teal-500 bg-clip-text text-transparent"
          >
            Anagram Solver
          </h1>
          <form
            class="flex flex-col gap-4"
          >
            <div
              class="mt-4 flex justify-center"
            >
              <input
                class="rounded-md lg:rounded-l-md lg:rounded-r-none border-slate-400"
                data-testid="input-test"
                id="anagram"
                name="anagram"
                placeholder="Your text ..."
                type="text"
                value=""
              />
              <button
                class="hidden lg:inline-flex items-center rounded-r-md border border-l-0 border-gray-300 bg-gray-50 py-3 px-3 text-sm text-gray-500"
                type="submit"
              >
                <svg
                  class="w-6 h-6"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21 21l-5.197-5.197m0 0A7.5 7.5 0 105.196 5.196a7.5 7.5 0 0010.607 10.607z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </button>
            </div>
            <button
              class="mx-auto px-5 lg:hidden font-semibold py-1 rounded-md bg-slate-200 border border-slate-500"
              type="submit"
            >
              Search
            </button>
          </form>
        </div>
      </div>
    </section>
    <footer
      class="absolute max-w-5xl px-4 flex bottom-0 justify-center sm:justify-end w-full"
    >
      <h5
        class="text-sm pt-8 pb-3"
      >
        Made with ðŸ’œ by 
        <a
          class="text-blue-900 underline underline-offset-2"
          href="https://twitter.com/_tittoh"
        >
          Tittoh
        </a>
      </h5>
    </footer>
  </main>
</DocumentFragment>
`;
